src/MediaBundleForm.php

```diff
@@ -101,6 +139,28 @@ public function form(array $form, FormStateInterface $form_state) {
   /**
    * {@inheritdoc}
    */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+
+    // Let the selected plugin validate its settings.
+    $plugin = $this->entity->getType()->getPluginId();
```

Comment:
After discussing this we found that $this->entity already contains the newly built entity...so no need to change this.